spec/ParseObject.spec.js

```diff
@@ -2055,4 +2055,33 @@ describe('Parse.Object testing', () => {
     const object = new Parse.Object('CloudCodeIsNew');
     await object.save();
   });
+
+  fit('should can custom objectId', async () => {
+    await reconfigureServer({
+      allowCustomObjectId: true,
+    });
+
+    Parse.allowCustomObjectId = true;
+
+    const body = {
+      name: 'SaveAudio',
+      data: '{"objectId":"66541dbdffead844f1339e3c","createdAt":"2024-05-27T05:44:29.504Z","updatedAt":"2024-05-27T05:44:29.504Z"}',
+    };
+
+    const name = body.name;
+    const data = JSON.parse(body.data);
+
+    const testObject = new Parse.Object(name);
+    testObject.id = data.objectId;
+
+    const entries = Object.entries(data);
+    for (let i = 0; i < entries.length; i++) {
+      const key = entries[i][0];
+      const value = entries[i][1];
+      testObject.set(key, value);
+    }
+
+    console.log(testObject.toJSON());
+    await testObject.save(null, { useMasterKey: true });
```

Comment:
At the end of the test you may want to fetch the object by the custom ID to ensure it has been created correctly in the DB.